CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.12)
+project(app-framework)
+
+set(CMAKE_CXX_STANDARD 17)
+
+include_directories(${CMAKE_SOURCE_DIR}/app-framework/include)
+include_directories(${CMAKE_SOURCE_DIR}/app-framework-base/include)
+
+add_subdirectory(apps)
+add_subdirectory(doc)
+add_subdirectory(include)
+add_subdirectory(scripts)
+add_subdirectory(src)
+add_subdirectory(test)
+add_subdirectory(unittest)
```

Comment:
Is it possible to:

* extract this boiler plate into a file that a package `#include`s in some way (instead of copy-pastes) 
* to not supply,eg `apps/` but to have `add_subdirectory()` not fail?

What's I'm hoping CMake can do is like what `waf` lets us.  A very minimal per-package build file.  Eg:

https://github.com/WireCell/wire-cell-toolkit/blob/master/img/wscript_build

It names itself and any dependencies and that's it.